### ISSUE TITLE
Github actions integration

### ISSUE THREAD

**Issue by josuablejeru** [sharkdp/bat#978]
Is there any interest in using Github actions for this Project? I had a nice experience using them and I would set up one for testing and build automation.

**Comment by sharkdp** [comment]
> Is there any interest in using Github actions for this Project?

Yes, there is! :smile: 

@rivy has recently upgraded the CICD system to GitHub Actions in another Rust project of mine (https://github.com/sharkdp/pastel, https://github.com/sharkdp/pastel/pull/120), and I really like how it nicely integrates with the rest of the UI.

That being said, I will put out the following disclaimer:
- The `bat` deployment system has grown over the last months and is a bit more involved than the one for `pastel`.
- The current TravisCI/Appveyor setup could probably be a bit more stable, but it has actually served us quite well in the last months. CI was up most of the time and deployment worked without any issues. For this reason, I am personally not motivated to do any work on switching to a new system. If we really want to make the switch, I would require that someone else tests this thoroughly on a separate repository first

**Comment by rivy** [comment]
Actually, the GHA CICD that I implemented for `pastel` will likely work here with just a few, fairly minor changes and some Windows bug squashing/test fixes. I pushed a run that was successful for everything except the windows builds (see <https://github.com/rivy/rust.bat/actions>).

It auto-deploys to GitHub Releases, as well.

I'll post a draft PR, but it will take me a little while to get everything fixed and polished. (I've got a fairly large reorganization PR that I'm working on for the uutils [coreutils](https://github.com/uutils/coreutils) project that I need to finish first.)

I'll be happy to look at incorporating any build processes that I'm missing, as well.
